ui/desktop/electron-app/src/services/rdp-client-manager.js

```diff
@@ -0,0 +1,177 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: BUSL-1.1
+ */
+
+const { spawn } = require('child_process');
+const { shell } = require('electron');
+const fs = require('fs');
+const which = require('which');
+const { isMac, isWindows } = require('../helpers/platform.js');
+const store = require('./electron-store-manager');
+const log = require('electron-log/main');
+
+// RDP Client Configuration
+const RDP_CLIENTS = [
+  {
+    value: 'mstsc',
+    isAvailable: async () => {
+      if (!isWindows()) return false;
+      try {
+        const mstscPath = await which('mstsc', { nothrow: true });
+        return Boolean(mstscPath);
+      } catch {
+        return false;
+      }
+    },
+  },
+  {
+    value: 'windows-app',
+    isAvailable: async () => {
+      if (!isMac()) return false;
+      try {
+        // Check for Windows App
+        if (fs.existsSync('/Applications/Windows App.app')) {
+          return true;
+        }
+
+        // Fallback: Use mdfind for Microsoft Remote Desktop
+        return new Promise((resolve) => {
+          const mdfind = spawn('mdfind', [
+            'kMDItemCFBundleIdentifier == "com.microsoft.rdc.macos"',
+          ]);
+          let output = '';
+
+          mdfind.stdout.on('data', (data) => {
+            output += data.toString();
+          });
+
+          mdfind.on('close', (code) => {
+            const found = code === 0 && output.trim().length > 0;
+            resolve(found);
+          });
+
+          mdfind.on('error', () => resolve(false));
+        });
+      } catch {
+        return false;
+      }
+    },
+  },
+  {
+    value: 'none',
+    isAvailable: () => true,
+  },
+];
+
+class RdpClientManager {
+  /**
+   * Gets all available RDP clients on the current system
+   * @returns {Promise<Array>} Array of available RDP clients with value
+   */
+  async getAvailableRdpClients() {
+    const available = [];
+    for (const client of RDP_CLIENTS) {
+      if (await client.isAvailable()) {
+        available.push({
+          value: client.value,
+        });
```

Comment:
How come we return an object that just has value instead of an array of strings?